homeassistant/components/meteo_france/sensor.py

```diff
@@ -213,3 +213,14 @@ def device_state_attributes(self):
             **readeable_phenomenoms_dict(self.coordinator.data.phenomenons_max_colors),
             ATTR_ATTRIBUTION: ATTRIBUTION,
         }
+
+    async def async_will_remove_from_hass(self):
+        """Unsubscribe from needed data on remove."""
+        await super().async_will_remove_from_hass()
+
+        self.coordinator.hass.data[DOMAIN][self._dept_code][
+            COORDINATOR_ALERT_ADDED
+        ] = False
```

Comment:
You should handle this in the `__init__.py:async_unload_entry`.

(and `hass` should be available in `self` because the `Entity` has it)

To do that, I think we should modify the `hass.data[DOMAIN]` as :
```
    hass.data[DOMAIN][entry.entry_id] = {
        COORDINATOR_FORECAST: coordinator_forecast,
        COORDINATOR_RAIN: coordinator_rain,
        COORDINATOR_ALERT: coordinator_alert,
    }
    ...
    hass.data[DOMAIN][department] = True
```